server/tests/src/test/java/org/infinispan/server/functional/RestCacheManagerResource.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.server.functional;
+
+import static org.infinispan.server.security.Common.sync;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+import org.infinispan.client.rest.RestCacheManagerClient;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.server.test.junit4.InfinispanServerRule;
+import org.infinispan.server.test.junit4.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+/**
+ * @author Ryan Emerson
+ * @since 11.0
+ */
+public class RestCacheManagerResource {
```

Comment:
What is the difference to https://github.com/infinispan/infinispan/blob/7470fe04cf105cf545d9ab7cf2889238de9a68d0/server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java  ?

Reply:
These tests can't be ran against the images. I came up with this test originally for the native server as there was an issue serializing the various inner classes of `CacheManagerResource` https://github.com/infinispan/infinispan-quarkus/pull/13. I figured it made sense to also have it in the main repo as these test are ran against various images.

Reply:
> These tests can't be ran against the images

I guess only tests inside ```server/test``` are run against the image? All the Rest resources are tested inside ```rest```, except those that are intrinsically tied to the server (ServerResource).
I think we need a solution for this instead of duplicating tests

Reply:
> I guess only tests inside `server/test` are run against the image?

Yes.

> I think we need a solution for this instead of duplicating tests, otherwise there's no way of knowing what works and what not for the native image

It wasn't clear in my original message, but the `infinispan-quarkus` test will be removed if this test is merged here.

I agree that duplication between `server/rest` and `server/tests` isn't ideal, but I'm not sure of an alternative. Maybe in this case the test should just remain in the `infinispan-quarkus` repo? As the resources only fail if the native image does not add a reflection exception.

Reply:
I am thinking in decoupling the tests in ```rest``` from the embedded REST server allowing them to be executed in the ```server``` module with a "real" server

Reply:
That would be great from my perspective. It would make my life much easier if they reside in `server/tests`, but the main thing is that they utilise the server testdriver and use `InfinispanServerTestMethodRule` to create the clients. Hardcoded localhost IPs need to be avoided.

Reply:
Right. We will need to parametrise the creation of the client and the server :)

Reply:
Just create the client via the [TestClientDriver](https://github.com/infinispan/infinispan/blob/cbd50e3c2fdf7b50f097689d20add5ef3eded73d/server/testdriver/core/src/main/java/org/infinispan/server/test/api/TestClientDriver.java#L24) and you're all set :slightly_smiling_face:

Reply:
that could work. The tests in ```rest``` will allow pluggable clients and servers in the setup, just need to look into the security aspects: all tests are run with and without security enabled